### DISCUSSION TITLE
App is showing 7.3 and Terminal is showing 8.1

### DISCUSSION THREAD

**Discussion by giantsoup** [shivammathur/homebrew-php#759] [Q&A / Help] [answered]
I am updating my local php version to 8.1 from 7.3 and I was able to follow the install instructions to get the CLI to correctly show 8.1 when i run php -v. However, my laravel application is still showing 7.3.25 when i use the phpinfo() function.

I have reinstalled everything, updated, upgraded, and cleaned up brew, and made sure that nginx has been restarted, but my application still shows 7.3.25 running.

I am on an intel macbook pro, running macos Catalina.

One weird thing is that previously when i had been using the default Homebrew php versions i was able to run brew services list and see php running in there, but i dont see anything like a php service mentioned in this readme at all. Also when i run the services list now i can see php as not running and displaying an error 19968 which I have never seen before.

```
mysql@5.7 started      tayloroyer ~/Library/LaunchAgents/homebrew.mxcl.mysql@5.7.plist
nginx     started      tayloroyer ~/Library/LaunchAgents/homebrew.mxcl.nginx.plist
php       error  19968 tayloroyer ~/Library/LaunchAgents/homebrew.mxcl.php.plist
```

Any help would be great. I had seen multiple articles recommending using this php tap instead of the default Homebrew php tap but now its not updating to 8.1 as expected. Let me know if I need to provide any other information.

**Comment by shivammathur** [comment]
Please try the following
1) Run `brew services cleanup` and try restarting php `brew services restart php`
2) Next, check if php is running or not using `brew services`
3) If it still says error, please provide the `php-fpm` log.
``` 
cat $(brew --prefix)/var/log/php-fpm.log
```

**Reply by giantsoup** [reply]
Amazing! `brew services cleanup` worked perfectly. Thank you for that. I don't know how i didn't find that recommendation while googling for that error but that worked exactly as expected.

EDIT: Added an additional note about this not working for trying to install php7.4.

**Comment by giantsoup** [comment]
So that solved the problem for the php error and I did see 8.1.2 being used in the app. But now i tried to install and switch to 7.4 and its doing the same thing again where its showing as 7.4 in the terminal after following the instructions, but in the app its still showing 8.1 in the app.

Here are the steps i followed to switch to 7.4:
- `brew install shivammathur/php/php@7.4`
- `brew link --overwrite --force shivammathur/php/php@7.4`
- `brew services restart php`
- `brew services restart nginx`

Then after all that i correctly see the terminal showing php 7.4 running but when i check phpinfo() in my app its still showing 8.1.2.

`brew services list` is showing everying green and running:
```
mysql@5.7 started tayloroyer ~/Library/LaunchAgents/homebrew.mxcl.mysql@5.7.plist
nginx     started tayloroyer ~/Library/LaunchAgents/homebrew.mxcl.nginx.plist
php       started tayloroyer ~/Library/LaunchAgents/homebrew.mxcl.php.plist
```

So is there a step I'm missing? I'm following the instructions exactly as stated in the README but its not working as intended. Any help would be greatly appreciated.

**Comment by shivammathur** [comment, marked as answer]
There should be a service for `php@7.4` when you run `brew services`.
If it is not there try reinstalling `php@7.4`

```zsh
brew reinstall shivammathur/php/php@7.4
```

Then switch to `php@7.4` service.

```zsh
brew services stop php
brew services start php@7.4
```

**Reply by giantsoup** [reply]
Ok yeah, I was not stopping the default process and starting the new one. The README does not mention this step and it is critical in making it work properly. I understand this may seem like an obvious step in the process but for people like me who do not use these types of things it would be very helpful to have this last step included.

Thank you for your help.

**Reply by shivammathur** [reply]
I will update the README

**Comment by estudio7** [comment]
HI!!
im stuck :(

php   error  19968 migueloduber ~/Library/LaunchAgents/homebrew.mxcl.php.plist

[14-Jan-2023 02:58:11] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[14-Jan-2023 02:58:11] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[14-Jan-2023 02:58:11] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[14-Jan-2023 02:58:11] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[14-Jan-2023 02:58:11] ERROR: unable to bind listening socket for address '127.0.0.1:9000': Address already in use (48)
[14-Jan-2023 02:58:11] ERROR: unable to bind listening socket for address '127.0.0.1:9000': Address already in use (48)
[14-Jan-2023 02:58:11] ERROR: FPM initialization failed
[14-Jan-2023 02:58:11] ERROR: FPM initialization failed

**Reply by shivammathur** [reply]
This might help
https://stackoverflow.com/questions/57975241/php-fpm-error-unable-to-bind-listening-socket-for-address-127-0-0-19000-ad
